test/tools/auth0/handlers/organizations.tests.js

```diff
@@ -1,14 +1,13 @@
+import { PromisePoolExecutor } from 'promise-pool-executor';
+
 const { expect } = require('chai');
 const organizations = require('../../../../src/tools/auth0/handlers/organizations');
 
-const pool = {
-  addEachTask: (data) => {
-    if (data.data && data.data.length) {
-      data.generator(data.data[0]);
-    }
-    return { promise: () => null };
-  },
-};
+const pool = new PromisePoolExecutor({
+  concurrencyLimit: 3,
+  frequencyLimit: 1000,
+  frequencyWindow: 1000, // 1 sec
+});
```

Comment:
The previous `pool` mock was swallowing errors which prevented assertions from being enforced. @ubenzer This fixes the weird behavior you saw.